File: code/bit-manipulation/sum_equals_xor/sum_equals_xor.java
/*
 * Given a positive integer n, find count of positive integers i such that 0 <= i <= n and n+i = n^i
 * Part of Cosmos by OpenGenus Foundation
 * More info: http://www.geeksforgeeks.org/equal-sum-xor/
 * 
 * */

import java.util.*;

public class sum_equals_xor {
  public static void main (String[] args) {
    System.out.println(solve(12));
  }
  
  public static long solve(long n) {
    long unset_binary_rep_n = 0;
    while(n > 0) {
      if((n & 1) == 0) 
        unset_binary_rep_n++;
      n = n >> 1;
    }
    unset_binary_rep_n = (long) Math.pow(2,unset_binary_rep_n);
    return unset_binary_rep_n;
  }
}

File: code/bit-manipulation/sum_equals_xor/sum_equals_xor.py
# Part of Cosmos by OpenGenus Foundation 

# Counts the number of values between 0 and a given number that satisfy the condition x+n = x^n (0<=x<=n)

# Works for both python 2 and python 3

n=int(input())
print (int(pow(2,(bin(n)[2:]).count('0'))))
